src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -239,12 +239,17 @@ protected virtual Task HandleInvalidContentTypeErrorAsync(HttpContext context)
         protected virtual Task HandleInvalidHttpMethodErrorAsync(HttpContext context)
             => WriteErrorResponseAsync(context, $"Invalid HTTP method. Only GET and POST are supported. {DOCS_URL}", HttpStatusCode.MethodNotAllowed);
 
-        protected virtual Task<ExecutionResult> ExecuteRequestAsync(GraphQLRequest gqlRequest, IDictionary<string, object> userContext, IGraphQLExecuter<TSchema> executer, IServiceProvider requestServices, CancellationToken token)
-            => executer.ExecuteAsync(
-                gqlRequest,
-                userContext,
-                requestServices,
-                token);
+        protected virtual Task<ExecutionResult> ExecuteRequestAsync(GraphQLRequest gqlRequest, IDictionary<string, object> userContext, IDocumentExecuter<TSchema> executer, IServiceProvider requestServices, CancellationToken token)
```

Comment:
```suggestion
        protected virtual Task<ExecutionResult> ExecuteRequestAsync(
        GraphQLRequest gqlRequest,
        IDictionary<string, object> userContext,
        IDocumentExecuter<TSchema> executer,
        IServiceProvider requestServices,
        CancellationToken token)
```

Reply:
Before changes options were copied here:
```
  protected virtual ExecutionOptions GetOptions(GraphQLRequest request, IDictionary<string, object> context, IServiceProvider requestServices, CancellationToken cancellationToken)
        {
            var opts = new ExecutionOptions
            {
                Schema = Schema,
                OperationName = request.OperationName,
                Query = request.Query,
                Variables = request.Variables,
                Extensions = request.Extensions,
                UserContext = context,
                CancellationToken = cancellationToken,
                ComplexityConfiguration = _options.ComplexityConfiguration,
                EnableMetrics = _options.EnableMetrics,
                UnhandledExceptionDelegate = _options.UnhandledExceptionDelegate,
                MaxParallelExecutionCount = _options.MaxParallelExecutionCount,
                RequestServices = requestServices,
            };
```

What are recommendations now?

Reply:
Options are configured by the `IGraphQLBuilder.ConfigureExecution` method.  Only the request (query, operation name, variables and extensions), service provider, and cancellation token are required.  The schema is provided by `IDocumentExecuter<TSchema>`.  The user context is provided here because for WebSocket connections, the user context needs to only be built once and not be built subsequent times.  Additional validation rules can be provided here, such as a validation use that I wrote for `GraphQL.AspNetCore3` to prevent GET calls from executing mutations.  The server project has no such use case.  Any other validation rules, complexity configuration, or anything else can be configured via `IGraphQLBuilder`.

Reply:
I'm not sure if I answered your question.